scripts/lib/CIME/case.py

```diff
@@ -457,10 +456,8 @@ def _set_compset(self, compset_name, files):
                 compsets = Compsets(compsets_filename)
                 match, compset_alias, science_support = compsets.get_compset_match(name=compset_name)
                 if match is not None:
-                    self._compsetsfile = compsets_filename
                     self._compsetname = match
-                    self.set_lookup_value("COMPSETS_SPEC_FILE" ,
-                                   files.get_value("COMPSETS_SPEC_FILE", {"component":component}, resolved=False))
+                    self.set_lookup_value("COMPSETS_SPEC_FILE", compsets_filename)
```

Comment:
Nope, I tried that - its needed in both places - perhaps you could write a test to boost your confidence?

Reply:
It would need to be a case which is not dependent on an external component.   I think the
ADESP tests might do the trick.

Reply:
Can you describe what doesn't work when that's removed? I just tried some manual tests with this diff:

```diff
diff --git a/scripts/lib/CIME/case.py b/scripts/lib/CIME/case.py
index b9a36ba72..07452e15f 100644
--- a/scripts/lib/CIME/case.py
+++ b/scripts/lib/CIME/case.py
@@ -457,7 +457,6 @@ def _set_compset(self, compset_name, files):
                 match, compset_alias, science_support = compsets.get_compset_match(name=compset_name)
                 if match is not None:
                     self._compsetname = match
-                    self.set_lookup_value("COMPSETS_SPEC_FILE", compsets_filename)
                     logger.info("Compset longname is {}".format(match))
                     logger.info("Compset specification file is {}".format(compsets_filename))
                     return compset_alias, science_support
```

I tried creating some cases with aliases and with longname (a compset that doesn't have an alias). My tests worked and gave the same final xml files as I got before introducing that change.

If it's true that that line is needed, then it seems that something is still wrong, because it implies that compsets with aliases are being treated differently in some way from compsets without aliases – which is what we want to avoid.

Reply:
The problem was with compsets whose primary component was CPL.  I think that the 
PRE.f19_f19.ADESP_TEST.cheyenne_intel demonstrates.